WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -27,192 +21,41 @@ namespace WalletWasabi.Fluent.ViewModels.CoinControl;
 	NavigationTarget = NavigationTarget.DialogScreen)]
 public partial class SelectCoinsDialogViewModel : DialogViewModelBase<IEnumerable<SmartCoin>>
 {
-	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel, IEnumerable<SmartCoin> selectedCoins)
+	private readonly TransactionInfo _transactionInfo;
+	private readonly WalletViewModel _walletViewModel;
+
+	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel, IEnumerable<SmartCoin> selectedCoins, TransactionInfo transactionInfo)
 	{
-		var pockets = walletViewModel.Wallet.GetPockets();
-		var pocketItems = CreatePocketItems(pockets);
-		
-		SyncSelectedItems(pocketItems, selectedCoins);
+		_walletViewModel = walletViewModel;
+		_transactionInfo = transactionInfo;
 
-		var pocketColumn = PocketColumn();
-		Source = new HierarchicalTreeDataGridSource<CoinControlItemViewModelBase>(pocketItems)
-		{
-			Columns =
-			{
-				ChildrenColumn(),
-				IndicatorsColumn(),
-				AmountColumn(),
-				AnonymityScoreColumn(),
-				pocketColumn
-			}
-		};
+		CoinSelector = new CoinSelectorViewModel(walletViewModel, selectedCoins);
 
-		Source.SortBy(pocketColumn, ListSortDirection.Descending);
-		Source.RowSelection!.SingleSelect = true;
+		var requiredAmount = CoinSelector.SelectedCoinsChanged.Select(GetRequiredAmount);
+		var selectedAmount = CoinSelector.SelectedCoinsChanged.Select(c => new Money(c.Sum(x => x.Amount)));
+		var remainingAmount = selectedAmount.CombineLatest(requiredAmount, (selected, required) => required - selected);
 
-		SetupCancel(false, true, false);
+		EnoughSelected = remainingAmount.Select(remaining => remaining <= Money.Zero).ReplayLastActive();
```

Comment:
We are not dealing with amounts in the UI currently. The return value of `TryBuildTransactionWithoutPrevTx` could be the value of `EnoughSelected`.